docs/book/component-guide/model-deployers/bentoml.md

```diff
@@ -98,13 +117,85 @@ def bento_builder_pipeline():
+# 1. use the step context to get the output artifact uri
+context = get_step_context()
+
+# 2. you can save the model and bento uri as part of the bento labels
+labels = labels or {}
+labels["model_uri"] = model.uri
+labels["bento_uri"] = os.path.join(
+    context.get_output_artifact_uri(), DEFAULT_BENTO_FILENAME
+)
+
+# 3. Load the model from the model artifact
+model = load_artifact_from_response(model)
+
+# 4. Save the model to a BentoML model based on the model type
+try:
+    module = importlib.import_module(f".{model_type}", "bentoml")
+    module.save_model(model_name, model, labels=labels)
+except importlib.metadata.PackageNotFoundError:
+    bentoml.picklable_model.save_model(
+        model_name,
+        model,
+    )
+
+# 5. Build the BentoML bundle. You can use any of the parameters supported by the bentos.build function.
+bento = bentos.build(
+    service=service,
+    models=[model_name],
+    version=version,
+    labels=labels,
+    description=description,
+    include=include,
+    exclude=exclude,
+    python=python,
+    docker=docker,
+    build_ctx=working_dir or source_utils.get_source_root(),
+)
+
+return bento
+
+```
+You can now use this bento in any way you see fit.
 
 ### ZenML BentoML Deployer step
 
-We have now built our bento bundle, and we can use the built-in `bentoml_model_deployer_step` to deploy the bento bundle to our local HTTP server. The following example shows how to call the built-in bento deployer step within a ZenML pipeline.
+We have now built our bento bundle, and we can use the built-in `bentoml_model_deployer_step` to deploy the bento bundle to our local HTTP server or to a containerized service running in your local machine. 
+
+{% hint style="info" %}
+The `bentoml_model_deployer_step` can only be used in a local environment. But in the case of using containerized deployment, you can use the Docker image created by the `bentoml_model_deployer_step` to deploy your model to a remote environment. It is automatically pushed to your ZenML Stack's container registry.
+{% endhint %}
+
+**Local deployment**
+
+The following example shows how to use the `bentoml_model_deployer_step` to deploy the bento bundle to a local HTTP server.
+
+```python
+from zenml import pipeline, step
+from zenml.integrations.bentoml.steps import bentoml_model_deployer_step
+
+@pipeline
+def bento_deployer_pipeline():
+    bento = ...
+    deployed_model = bentoml_model_deployer_step(
+        bento=bento
+        model_name="pytorch_mnist",  # Name of the model
+        port=3001,  # Port to be used by the http server
+    )
+```
+
+**Containerized deployment**
+
+The following example shows how to use the `bentoml_model_deployer_step` to deploy the bento bundle to a [containerized service](https://docs.bentoml.org/en/latest/guides/containerization.html) running in your local machine. Make sure you have the `docker` CLI installed on your local machine to be able to build an image and deploy the containerized service.
```

Comment:
I think the deployer step is a bit more advanced because it deals with ZenML Services which is a concept internal to ZenML and the users shouldn't have to worry about learning how it works just to deploy a model. We already do have a large range of customization that is offered while deploying. wdyt?